octobot_trading/orders/types/trailing/trailing_stop_order.py

```diff
@@ -16,40 +16,156 @@
 import asyncio
 from asyncio import wait_for
 
+from octobot_commons.logging.logging_util import get_logger
+
 from octobot_trading.data.order import Order
-from octobot_trading.enums import TradeOrderSide
+from octobot_trading.enums import TradeOrderSide, TraderOrderType
 
 
 class TrailingStopOrder(Order):
-    def __init__(self, trader):
-        super().__init__(trader)
-        self.side = TradeOrderSide.SELL
+    UNINITIALIZED_TRAILING_PERCENT = -1
+    DEFAULT_TRAILING_PERCENT = 5
+
+    def __init__(self, trader, side=TradeOrderSide.SELL, trailing_percent=UNINITIALIZED_TRAILING_PERCENT):
+        super().__init__(trader, side=side)
         self.trailing_stop_price_hit_event = None
         self.trailing_price_hit_event = None
-        self.wait_for_hit_event_task = None
+        self.wait_for_stop_price_hit_event_task = None
+        self.wait_for_price_hit_event_task = None
+        self.trailing_percent = trailing_percent
 
     async def update_order_status(self, force_refresh=False):
         if not self.trader.simulate and (not self.is_synchronized_with_exchange or force_refresh):
             await self.default_exchange_update_order_status()
 
-        if self.trailing_stop_price_hit_event is None:
-            self.trailing_stop_price_hit_event = self.exchange_manager.exchange_symbols_data.\
-                get_exchange_symbol_data(self.symbol).price_events_manager.\
-                add_event(self.origin_price, self.creation_time, self.side is TradeOrderSide.SELL)
+        await self._reset_events(self.origin_price, self.creation_time, force_reset=False)
+
+    async def set_trailing_percent(self, trailing_percent):
+        """
+        Set trailing percent and reset event and tasks
+        :param trailing_percent: the new trailing percent
+        """
+        self.trailing_percent = trailing_percent
+        await self._reset_events(self.origin_price, self.creation_time, force_reset=True)
```

Comment:
is it OK to call a cythonized positional argument as a named argument ? I remember we had issues about this

Reply:
Hum... I don't remember, I'll have a look at this method in compiled version

Reply:
`force_reset` doesn't exist anymore :wink:

Reply:
perfect :)